frontend/unit.js

```diff
@@ -0,0 +1,39 @@
+class Unit {
+    constructor(texture, grid, y, x, params) {
+        this.sprite = new PIXI.Sprite(texture);
+        this.grid = grid;
```

Comment:
I will argue with this solution

It feels illogical that units can change environment (grid). When grid knows about unit and unit knows about grid bad things start happening. Program becomes overgrown with excessive dependencies. And data duplication (grid.hex[y][x] and unit.x, unit.y) leads to divergence of this data in future

That's why I suggest remove grid and coordinates from unit and left units referable from hexes (as you already did)

Reply:
ok, agree

Reply:
Second @bronvic